examples/pytorch/graphsage/dist/train_dist.py

```diff
@@ -284,13 +234,17 @@ def main(args):
         g.rank(), len(train_nid), len(np.intersect1d(train_nid.numpy(), local_nid)),
         len(val_nid), len(np.intersect1d(val_nid.numpy(), local_nid)),
         len(test_nid), len(np.intersect1d(test_nid.numpy(), local_nid))))
+    del local_nid
     if args.num_gpus == -1:
         device = th.device('cpu')
     else:
         dev_id = g.rank() % args.num_gpus
         device = th.device('cuda:'+str(dev_id))
-    labels = g.ndata['labels'][np.arange(g.number_of_nodes())]
-    n_classes = len(th.unique(labels[th.logical_not(th.isnan(labels))]))
+    n_classes = args.n_classes
+    if n_classes == 0:
+        labels = g.ndata['labels'][np.arange(g.number_of_nodes())]
```

Comment:
Is `g.ndata['labels'][np.arange(g.number_of_nodes())]` different from `g.ndata['labels']`?

Reply:
No, we need real tensor instead of `DistTensor`. Or below error is thrown:
```
Traceback (most recent call last):                                                                                                 
  File "train_dist.py", line 436, in <module>
    main(args)                                                                                                                 
  File "train_dist.py", line 364, in main                                                                                          
    n_classes = len(th.unique(labels[th.logical_not(th.isnan(labels))]))        
TypeError: isnan(): argument 'input' (position 1) must be Tensor, not DistTensor 
```